WeakAuras/RegionTypes/Text.lua

```diff
@@ -169,76 +174,124 @@ local function modify(parent, region, data)
       end
+      local formatters = Private.CreateFormatters(self.displayText, getter)
+      UpdateText = function()
+        local textStr = self.displayText;
+        textStr = Private.ReplacePlaceHolders(textStr, self, nil, false, formatters);
+        if (textStr == nil or textStr == "") then
+          textStr = " ";
+        end
 
-      SetText(textStr)
+        SetText(textStr)
+      end
     end
-  end
 
-  local customTextFunc = nil
-  if(Private.ContainsCustomPlaceHolder(data.displayText) and data.customText) then
-    customTextFunc = WeakAuras.LoadFunction("return "..data.customText)
-  end
+    local customTextFunc = nil
+    if(Private.ContainsCustomPlaceHolder(self.displayText) and data.customText) then
+      customTextFunc = WeakAuras.LoadFunction("return "..data.customText)
+    end
 
-  local Update
-  if customTextFunc then
-    if UpdateText then
+    local Update
+    if customTextFunc and UpdateText then
       Update = function()
-        region.values.custom = Private.RunCustomTextFunc(region, customTextFunc)
+        self.values.custom = Private.RunCustomTextFunc(self, customTextFunc)
         UpdateText()
       end
+    else
+      Update = UpdateText or function() end
     end
-  else
-    Update = UpdateText or function() end
-  end
 
-  local TimerTick
-  if Private.ContainsPlaceHolders(data.displayText, "p") then
-    TimerTick = UpdateText
-  end
+    local TimerTick
+    if Private.ContainsPlaceHolders(self.displayText, "p") then
+      TimerTick = UpdateText
+    end
 
-  local FrameTick
-  if customTextFunc and data.customTextUpdate == "update" then
-    FrameTick = function()
-      region.values.custom = Private.RunCustomTextFunc(region, customTextFunc)
-      UpdateText()
+    local FrameTick
+    if customTextFunc and data.customTextUpdate == "update" then
+      if Private.ContainsCustomPlaceHolder(self.displayText) then
+        FrameTick = function()
+          self.values.custom = Private.RunCustomTextFunc(self, customTextFunc)
+          UpdateText()
+        end
+      end
     end
-  end
 
-  region.Update = Update
-  region.FrameTick = FrameTick
-  region.TimerTick = TimerTick
-  if TimerTick then
-    region.subRegionEvents:AddSubscriber("TimerTick", region, true)
-  else
-    region.subRegionEvents:RemoveSubscriber("TimerTick", region)
-  end
+    self.Update = Update
+    self.FrameTick = FrameTick
+    self.TimerTick = TimerTick
 
-  if FrameTick then
-    region.subRegionEvents:AddSubscriber("FrameTick", region)
-  else
-    region.subRegionEvents:RemoveSubscriber("FrameTick", region)
+    if not self.UpdateText then
+      local textStr = self.displayText
+      textStr = textStr:gsub("\\n", "\n");
+      SetText(textStr)
+    end
   end
 
-  if not UpdateText then
-    local textStr = data.displayText
-    textStr = textStr:gsub("\\n", "\n");
-    SetText(textStr)
+  function region:ConfigureSubscribers()
+    local visible = self:IsShown()
+    if self.Update then
+      if visible then
```

Comment:
That can't be right, there's no code to register the subscribers if the text becomes visible.

I think this check is bogus for texts.